designate/central/service.py

```diff
@@ -3101,8 +3106,62 @@ def _import(self, context, zone_import, request_body, pool_id=''):
 
         # If the zone import was valid, create the zone
         if zone_import.status != 'ERROR':
+
             try:
-                zone = self.create_zone(context, zone)
+                LOG.debug(f'Force enabled {force}')
+                if force:
+                    try:
+                        zone_object = self.find_zone(
+                            context,
+                            {'name': zone.name}
+                        )  # without recordsets object
+                        LOG.debug(f'Founded zone {zone_object}')
+                    except exceptions.ZoneNotFound:
+                        zone = self.create_zone(context, zone)
+                    else:
+                        criterion = {'zone_id': zone_object.id}
+                        rrsets = self.find_recordsets(context, criterion)
+                        # add rrset.name == zone.name
+                        # add tests for ns records with subdomain name
+                        actual_rrsets = dict()
+                        new_rrsets = dict()
+                        for rrset in rrsets:
```

Comment:
you can just pop the recordset you need out of array here and not create a new array.